src/openforms/submissions/tests/renderer/test_formio_integration.py

```diff
@@ -59,29 +60,28 @@ class SubmissionRendererIntegrationTests(TestCase):
     def setUpTestData(cls):
         super().setUpTestData()
 
-        submission = SubmissionFactory.create(
-            form__name="public name",
-            form__internal_name="internal name",
-            form__generate_minimal_setup=True,
-            form__formstep__form_definition__name="Stap 1",
-            form__formstep__form_definition__configuration={
-                "components": FORMIO_CONFIGURATION_COMPONENTS
-            },
-        )
-        step = SubmissionStepFactory.create(
-            submission=submission,
-            form_step=submission.form.formstep_set.get(),
-            data={
+        submission = SubmissionFactory.from_components(
+            components_list=FORMIO_CONFIGURATION_COMPONENTS,
+            submitted_data={
                 "input1": "first input",
                 "input2": "second input",
                 "amount": 1234.56,
                 "input4": "fourth input",
             },
         )
 
+        form = submission.form
+        form.name = "public name"
+        form.internal_name = "internal name"
+        form.save()
```

Comment:
you should be able to pass that to the `SubmissionFactory` as kwargs with double underscore, that's the main added value of factory-boy :-)